pkg/cluster/cluster.go

```diff
@@ -295,6 +295,8 @@ func (c *Cluster) run(stopC <-chan struct{}) {
 				break
 			}
 
+			garbagecollection.CollectDeadMemberServices(c.cluster.Name, c.cluster.Namespace, c.config.KubeCli, c.members, c.logger)
```

Comment:
I see. I do agree that coupling GC code here isn't a good idea.

This fix is naive. It doesn't consider performance impact too.

Achieving good perf and good code isn't my goal here. I would leave a TODO instead. We might need more complex algorithms to do GC though.